src/ImageSharp/Formats/Qoi/QoiDecoderCore.cs

```diff
@@ -0,0 +1,287 @@
+                            case QoiChunk.QoiOpLuma:
+                                byte diffGreen = (byte)(operationByte & 0b00111111),
+                                    currentGreen = (byte)((previousPixel.G + (diffGreen - 32)) % 256),
+                                    nextByte = (byte)stream.ReadByte(),
+                                    diffRedDG = (byte)(nextByte >> 4),
+                                    diffBlueDG = (byte)(nextByte & 0b00001111),
+                                    currentRed = (byte)((diffRedDG - 8 + (diffGreen - 32) + previousPixel.R) % 256),
+                                    currentBlue = (byte)((diffBlueDG - 8 + (diffGreen - 32) + previousPixel.B) % 256);
+                                readPixel = previousPixel with { R = currentRed, B = currentBlue, G = currentGreen };
+                                pixel.FromRgba32(readPixel);
+                                pixelArrayPosition = GetArrayPosition(readPixel);
+                                previouslySeenPixels[pixelArrayPosition] = readPixel;
+                                break;
+
+                            // Repeating the previous pixel 1..63 times
+                            case QoiChunk.QoiOpRun:
+                                byte repetitions = (byte)(operationByte & 0b00111111);
+                                if (repetitions is 62 or 63)
+                                {
+                                    ThrowInvalidImageContentException();
+                                }
+
+                                readPixel = previousPixel;
+                                pixel.FromRgba32(readPixel);
+                                for (int k = -1; k < repetitions; k++, j++)
+                                {
+                                    if (j == this.header.Width)
+                                    {
+                                        j = 0;
+                                        i++;
+                                    }
+
+                                    pixels[j, i] = pixel;
+                                }
+
+                                j--;
+                                continue;
+
+                            default:
+                                ThrowInvalidImageContentException();
+                                return;
+                        }
+
+                        break;
+                }
+
+                pixels[j, i] = pixel;
+                previousPixel = readPixel;
+            }
+        }
+
+        // Check stream end
+        for (int i = 0; i < 7; i++)
+        {
+            if (stream.ReadByte() != 0)
+            {
+                ThrowInvalidImageContentException();
+            }
+        }
+
+        if (stream.ReadByte() != 1)
+        {
+            ThrowInvalidImageContentException();
+        }
+    }
+
+    private static int GetArrayPosition(Rgba32 pixel) => ((pixel.R * 3) + (pixel.G * 5) + (pixel.B * 7) + (pixel.A * 11)) % 64;
```

Comment:
```suggestion
    [MethodImpl(MethodImplOptions.AggressiveInlining)]
    private static int GetArrayPosition(Rgba32 pixel) => (((pixel.R * 3) + (pixel.G * 5) + (pixel.B * 7) + (pixel.A * 11))) & 11;
```
Even better if you add a `Modulo64` method to our `Numerics` class.

Formula: https://stackoverflow.com/a/19760152

Reply:
Isn't it `& 63` to do a modulo 64?

Reply:
Yeah